lib/index.js

```diff
@@ -94,7 +94,33 @@ Insight.prototype.track = function () {
 	}).join('/');
 
 	// timestamp isn't unique enough since it can end up with duplicate entries
-	this._queue[Date.now() + ' ' + path] = path;
+	this._queue[Date.now() + ' ' + path] = {
+		path: path,
+		type: 'pageview'
+	};
+	this._save();
+};
+
+Insight.prototype.trackEvent = function (category, action, label, value) {
+	if (this.optOut) {
+		return;
+	}
+
+	if (this.trackingProvider !== 'google') {
+		throw new Error('event tracking is supported only for google-analytics');
+	}
+	if (!category || !action) {
+		throw new Error('event category and event action required');
+	}
+
+	// timestamp isn't unique enough since it can end up with duplicate entries
+	this._queue[Date.now() + ' ' + category + action + label + value] = {
```

Comment:
you are right. Can we use https://github.com/broofa/node-uuid for this? 
If you dont like an external dependency then we can do something simple like below to generate a UUID v4 (Found on SO [here](http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript) from author of the above plugin)
```
'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
    var r = Math.random()*16|0, v = c == 'x' ? r : (r&0x3|0x8);
    return v.toString(16);
});
```
if you are ok with either ill make the changes, will change for the track method as well for consistency